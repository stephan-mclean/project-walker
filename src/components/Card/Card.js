import React from "react";
import styles from "./Card.module.scss";

const Card = ({
  children,
  title,
  description,
  fullscreen,
  centerTitle,
  className
}) => {
  let cardClassName = `${
    styles.container
  } high-contrast-coloring margin-bottom ${className}`;
  if (fullscreen) {
    cardClassName += ` radius-bottom`;
  } else {
    cardClassName += " radius";
  }

  let titleDescClassName = `flex flex--column`;
  if (centerTitle) {
    titleDescClassName += ` flex--center-y`;
  }

  return (
    <div className={cardClassName}>
      {(title || description) && (
        <div className={titleDescClassName}>
          {title && <h6>{title}</h6>}
          {description && <p className="s2">{description}</p>}
        </div>
      )}

      {children}
    </div>
  );
};

export default Card;
