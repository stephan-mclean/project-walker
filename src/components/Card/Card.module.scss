.container {
  padding: var(--component-padding);
  box-shadow: var(--shadow-fg-tertiary-1-bottom-right-25);
  position: relative;
}
