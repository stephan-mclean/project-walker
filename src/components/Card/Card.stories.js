import React, { Fragment } from "react";
import { storiesOf } from "@storybook/react";

import Card from "./Card";

storiesOf("Components", module).add("Card", () => {
  return (
    <Fragment>
      <Card>Base</Card>

      <Card title="title">With title</Card>

      <Card title="title" centerTitle>
        With center title
      </Card>

      <Card title="title" description="description">
        With title and description
      </Card>

      <Card fullscreen>fullscreen</Card>

      <Card className="primary-coloring">With custom class</Card>
    </Fragment>
  );
});
