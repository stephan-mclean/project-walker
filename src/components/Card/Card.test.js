import React from "react";
import { shallow } from "enzyme";
import Card from "./Card";

describe("Testing the Card component", () => {
  it("Should match the snapshot", () => {
    const card = shallow(<Card>Card</Card>);
    expect(card).toMatchSnapshot();
  });

  it("Should render the correct content", () => {
    const card = shallow(<Card>With text</Card>);
    expect(card.text()).toEqual("With text");

    const cardWithChildren = shallow(
      <Card>
        <div>Child</div>
      </Card>
    );
    expect(cardWithChildren).toMatchSnapshot();
    expect(cardWithChildren.contains(<div>Child</div>)).toEqual(true);
  });

  it("Should render the title", () => {
    const card = shallow(<Card title="title" />);
    expect(card).toMatchSnapshot();
    expect(card.text()).toEqual("title");
  });

  it("Should render the description", () => {
    const card = shallow(<Card description="description" />);
    expect(card).toMatchSnapshot();
    expect(card.text()).toEqual("description");
  });

  it("Should allow using custom classes", () => {
    const card = shallow(<Card className="primary-coloring" />);
    expect(card).toMatchSnapshot();
    expect(card.hasClass("primary-coloring")).toEqual(true);
  });

  it("Should center the title", () => {
    const card = shallow(<Card title="Centered" centerTitle />);

    expect(
      card
        .find('div[data-testid="titleDescContainer"]')
        .hasClass("flex--center-y")
    ).toBeTruthy();
  });
});
