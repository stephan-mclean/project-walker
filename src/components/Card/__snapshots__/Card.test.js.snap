// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Card component Should allow using custom classes 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    className="primary-coloring"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        undefined,
      ],
      "className": "container high-contrast-coloring margin-bottom radius primary-coloring",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          undefined,
        ],
        "className": "container high-contrast-coloring margin-bottom radius primary-coloring",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Testing the Card component Should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card>
    Card
  </Card>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        "Card",
      ],
      "className": "container high-contrast-coloring margin-bottom radius ",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      "Card",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          "Card",
        ],
        "className": "container high-contrast-coloring margin-bottom radius ",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        "Card",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Testing the Card component Should render the correct content 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card>
    <div>
      Child
    </div>
  </Card>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <div>
          Child
        </div>,
      ],
      "className": "container high-contrast-coloring margin-bottom radius ",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Child",
        },
        "ref": null,
        "rendered": "Child",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <div>
            Child
          </div>,
        ],
        "className": "container high-contrast-coloring margin-bottom radius ",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Child",
          },
          "ref": null,
          "rendered": "Child",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Testing the Card component Should render the description 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    description="description"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="flex flex--column"
        >
          <p
            className="s2"
          >
            description
          </p>
        </div>,
        undefined,
      ],
      "className": "container high-contrast-coloring margin-bottom radius ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            <p
              className="s2"
            >
              description
            </p>,
          ],
          "className": "flex flex--column",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "description",
              "className": "s2",
            },
            "ref": null,
            "rendered": "description",
            "type": "p",
          },
        ],
        "type": "div",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="flex flex--column"
          >
            <p
              className="s2"
            >
              description
            </p>
          </div>,
          undefined,
        ],
        "className": "container high-contrast-coloring margin-bottom radius ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              <p
                className="s2"
              >
                description
              </p>,
            ],
            "className": "flex flex--column",
          },
          "ref": null,
          "rendered": Array [
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "description",
                "className": "s2",
              },
              "ref": null,
              "rendered": "description",
              "type": "p",
            },
          ],
          "type": "div",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Testing the Card component Should render the title 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    title="title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="flex flex--column"
        >
          <h6>
            title
          </h6>
        </div>,
        undefined,
      ],
      "className": "container high-contrast-coloring margin-bottom radius ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h6>
              title
            </h6>,
            undefined,
          ],
          "className": "flex flex--column",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "title",
            },
            "ref": null,
            "rendered": "title",
            "type": "h6",
          },
          undefined,
        ],
        "type": "div",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="flex flex--column"
          >
            <h6>
              title
            </h6>
          </div>,
          undefined,
        ],
        "className": "container high-contrast-coloring margin-bottom radius ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h6>
                title
              </h6>,
              undefined,
            ],
            "className": "flex flex--column",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "title",
              },
              "ref": null,
              "rendered": "title",
              "type": "h6",
            },
            undefined,
          ],
          "type": "div",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
